manifests/plugin/openscap/params.pp

```diff
@@ -9,4 +9,5 @@
   $reportsdir              = '/var/lib/foreman-proxy/openscap/reports'
   $failed_dir              = '/var/lib/foreman-proxy/openscap/failed'
   $corrupted_dir           = '/var/lib/foreman-proxy/openscap/corrupted'
+  $registered_proxy_name   = $foreman_proxy::registered_name
```

Comment:
This may result in unexpected results because it depends on compile order. If the params class is compiled without the proxy being compiled this will break. In the installed context this might lead to incorrect defaults. It's also another place to change the hostname which means the katello hostname change script will need modification. I'd recommend defaulting to `undef` and then in the main class you use `$proxy_name = pick($registered_proxy_name, $foreman_proxy::registered_name)`.